### ISSUE TITLE
LogicalResponse doesn't include lease information

### ISSUE THREAD

**Issue by kunickiaj** [BetterCloud/vault-java-driver#10]
It seems that LogicalResponse only includes the data portion of the REST reply.

It does not include important metadata such as:

```
"lease_id": "<secret backend>/creds/<role>/<uuid>",
"renewable": true,
"lease_duration": 3600,
```

This metadata is necessary in order for an application to honor the TTL for credentials.

I'm happy to submit a pull request if that seems reasonable. These would be provided as new properties of LogicalResponse.

**Comment by jmorefieldexpe** [comment]
I agree that this is essential for full use of this library!  Can we get this into the minor point release discussed in the other issues please?  If not open it up for contribution please :)

**Comment by steve-perkins** [comment]
I'm trying to determine whether these meta fields should be placed specifically in `LogicalResponse`, or if they're general enough to place in the `VaultResponse` base class.  In testing so far, Vault seems to return these meta fields for `Auth` operations as well as for `Logical` operations.  They aren't well documented... do you know whether they are indeed returned for ALL rest operations?

**Comment by kunickiaj** [comment]
I did notice that `renewable` and `lease_duration` are both also returned inside the auth object for all Auth operations AND in `VaultResponse`. Not sure yet if they're always the same in both places.

I'm also working on a Kotlin-based library for fun, where I've confirmed all the fields for Auth operations are the same except for auth.metadata which by auth backend.

**Comment by steve-perkins** [comment]
Now available in version 1.0.0.  This just went out, so it may take a few hours to sync with Maven Central.
